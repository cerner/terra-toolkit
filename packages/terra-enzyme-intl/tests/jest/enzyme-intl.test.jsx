import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, injectIntl } from 'react-intl';
import {
  shallowWithIntl,
  mountWithIntl,
  renderWithIntl,
  mockIntl,
} from '../../src';

const prettyPrintProps = props => JSON.stringify(props, undefined, 2);

const FormattedMessageSubject = props => (
  <div>
    <FormattedMessage id="TerraEnzymeIntl.helloWorld" />
    <FormattedMessage id="TerraEnzymeIntl.buttonText">
      {text => <button type="button">{text}</button>}
    </FormattedMessage>
    {prettyPrintProps(props)}
  </div>
);

const InjectIntlSubject = injectIntl(({ intl, ...props }) => (
  <div>
    {intl.formatMessage({ id: 'TerraEnzymeIntl.helloWorld' })}
    <button type="button">{intl.formatMessage({ id: 'TerraEnzymeIntl.buttonText' })}</button>
    {prettyPrintProps(props)}
  </div>
));

const WrappingComponentSubject = ({ bar, children }) => (
  <div>
    {prettyPrintProps(bar)}
    {children}
  </div>
);

WrappingComponentSubject.propTypes = {
  bar: PropTypes.string,
  children: PropTypes.node,
};

describe('shallowWithIntl', () => {
  describe('using FormattedMessage', () => {
    const subject = <FormattedMessageSubject foo="bar" />;
    const wrapper = shallowWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using injectIntl', () => {
    const subject = <InjectIntlSubject foo="bar" />;
    const wrapper = shallowWithIntl(subject).dive();

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('mountWithIntl', () => {
  describe('using FormattedMessage', () => {
    const subject = <FormattedMessageSubject foo="bar" />;
    const wrapper = mountWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using injectIntl', () => {
    const subject = <InjectIntlSubject foo="bar" />;
    const wrapper = mountWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('with a wrapping component', () => {
    const subject = <FormattedMessageSubject foo="bar" />;
    const wrapper = mountWithIntl(subject, { wrappingComponent: WrappingComponentSubject, wrappingComponentProps: { bar: 'foo' } });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('renderWithIntl', () => {
  describe('using FormattedMessage', () => {
    const subject = <FormattedMessageSubject foo="bar" />;
    const wrapper = renderWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using injectIntl', () => {
    const subject = <InjectIntlSubject foo="bar" />;
    const wrapper = renderWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('with a wrapping component', () => {
    const subject = <FormattedMessageSubject foo="bar" />;
    const wrapper = shallowWithIntl(subject, { wrappingComponent: WrappingComponentSubject, wrappingComponentProps: { bar: 'foo' } });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('mockIntl', () => {
  const subject = intl => intl.formatMessage({ id: 'TerraEnzymeIntl.foo' });
  const expected = 'TerraEnzymeIntl.foo';

  describe('when used in a method expecting the react-intl intl object', () => {
    it('should match the expected output', () => {
      expect(subject(mockIntl)).toEqual(expected);
    });
  });
});
