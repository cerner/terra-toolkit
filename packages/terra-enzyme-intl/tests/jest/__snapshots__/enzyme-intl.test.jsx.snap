// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mountWithIntl using FormattedMessage should match the snapshot 1`] = `
<FormattedMessageSubject
  foo="bar"
>
  <div>
    <FormattedMessage
      id="TerraEnzymeIntl.helloWorld"
    >
      TerraEnzymeIntl.helloWorld
    </FormattedMessage>
    <FormattedMessage
      id="TerraEnzymeIntl.buttonText"
    >
      <button
        type="button"
      >
        TerraEnzymeIntl.buttonText
      </button>
    </FormattedMessage>
    {
  "foo": "bar"
}
  </div>
</FormattedMessageSubject>
`;

exports[`mountWithIntl using injectIntl should match the snapshot 1`] = `
<injectIntl(Component)
  foo="bar"
>
  <Component
    foo="bar"
    intl={
      Object {
        "$t": [Function],
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "defaultRichTextElements": undefined,
        "fallbackOnEmptyString": true,
        "formatDate": [Function],
        "formatDateTimeRange": [Function],
        "formatDateToParts": [Function],
        "formatDisplayName": [Function],
        "formatList": [Function],
        "formatListToParts": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatNumberToParts": [Function],
        "formatPlural": [Function],
        "formatRelativeTime": [Function],
        "formatTime": [Function],
        "formatTimeToParts": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getDisplayNames": [Function],
          "getListFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralRules": [Function],
          "getRelativeTimeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "onError": [Function],
        "onWarn": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": undefined,
        "wrapRichTextChunksInFragment": undefined,
      }
    }
  >
    <div>
      TerraEnzymeIntl.helloWorld
      <button
        type="button"
      >
        TerraEnzymeIntl.buttonText
      </button>
      {
  "foo": "bar"
}
    </div>
  </Component>
</injectIntl(Component)>
`;

exports[`mountWithIntl with a wrapping component should match the snapshot 1`] = `
<FormattedMessageSubject
  foo="bar"
>
  <div>
    <FormattedMessage
      id="TerraEnzymeIntl.helloWorld"
    >
      TerraEnzymeIntl.helloWorld
    </FormattedMessage>
    <FormattedMessage
      id="TerraEnzymeIntl.buttonText"
    >
      <button
        type="button"
      >
        TerraEnzymeIntl.buttonText
      </button>
    </FormattedMessage>
    {
  "foo": "bar"
}
  </div>
</FormattedMessageSubject>
`;

exports[`renderWithIntl using FormattedMessage should match the snapshot 1`] = `
<div>
  TerraEnzymeIntl.helloWorld
  <button
    type="button"
  >
    TerraEnzymeIntl.buttonText
  </button>
  {
  "foo": "bar"
}
</div>
`;

exports[`renderWithIntl using injectIntl should match the snapshot 1`] = `
<div>
  TerraEnzymeIntl.helloWorld
  <button
    type="button"
  >
    TerraEnzymeIntl.buttonText
  </button>
  {
  "foo": "bar"
}
</div>
`;

exports[`renderWithIntl with a wrapping component should match the snapshot 1`] = `
<div>
  <MemoizedFormattedMessage
    id="TerraEnzymeIntl.helloWorld"
  />
  <MemoizedFormattedMessage
    id="TerraEnzymeIntl.buttonText"
  >
    <Component />
  </MemoizedFormattedMessage>
  {
  "foo": "bar"
}
</div>
`;

exports[`shallowWithIntl using FormattedMessage should match the snapshot 1`] = `
<div>
  <MemoizedFormattedMessage
    id="TerraEnzymeIntl.helloWorld"
  />
  <MemoizedFormattedMessage
    id="TerraEnzymeIntl.buttonText"
  >
    <Component />
  </MemoizedFormattedMessage>
  {
  "foo": "bar"
}
</div>
`;

exports[`shallowWithIntl using injectIntl should match the snapshot 1`] = `
<Component
  foo="bar"
  intl={
    Object {
      "$t": [Function],
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "defaultRichTextElements": undefined,
      "fallbackOnEmptyString": true,
      "formatDate": [Function],
      "formatDateTimeRange": [Function],
      "formatDateToParts": [Function],
      "formatDisplayName": [Function],
      "formatList": [Function],
      "formatListToParts": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatNumberToParts": [Function],
      "formatPlural": [Function],
      "formatRelativeTime": [Function],
      "formatTime": [Function],
      "formatTimeToParts": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getDisplayNames": [Function],
        "getListFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralRules": [Function],
        "getRelativeTimeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "onError": [Function],
      "onWarn": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": undefined,
      "wrapRichTextChunksInFragment": undefined,
    }
  }
/>
`;
